db/model/generator.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/model/generator.js
+ */
+const common = require('../helpers/common');
+const constants = require('../constants');
+const ValidationError = require('../dbErrors').ValidationError;
+
+const assoc = {};
+
+const generatorTemplateSchema = {
+  properties: {
+    name: {
+      description: 'GeneratorTemplate name associated with this generator',
+      type: 'string',
+      maxLength: constants.fieldlen.normalName,
+      pattern: constants.nameRegex,
+      required: true,
+    },
+    version: {
+      description: 'Version of the generatorTemplate associated with this ' +
+        'generator',
+      type: 'string',
+      pattern: constants.versionRegex,
```

Comment:
also accepts comparators, hyphen ranges, x-ranges, tilde ranges and caret ranges. See https://docs.npmjs.com/misc/semver for version grammar.